### PR TITLE
feat(EMI-1961): implement auction signals

### PR DESCRIPTION
**Pull request by rquartararo**
The type of this PR is: **Feature**

This PR solves [EMI-1961]

### Description

View [Figma](https://www.figma.com/design/22RFPSkUu97g0ska3azFL2/Social-proof-%26-urgency-signals?node-id=1799-9320&m=dev) for signal display logic

Adds
- Lot watcher counts
- Updated bid counts
- `Bidding closed` signal
- `Bidding live now` signal
- Bidding closing time timer
- Register date signal

Removes 
- Grid item badge
- Lot close info
- Auction name from lot detail cards

### Lot watcher counts + Bidding closing time timer
Screenshot 2024-08-23 at 7 12 36 AM

### `Bidding closed` signal
Screenshot 2024-08-23 at 7 12 22 AM

### `Bidding live now` signal
Screenshot 2024-08-23 at 7 11 39 AM

### Register date signal
Screenshot 2024-08-23 at 7 12 49 AM

### All together on one rail
Screenshot 2024-08-22 at 10 59 06 PM

### Lot detail cards
Screenshot 2024-08-22 at 11 27 52 PM

### Aucton detail page
Screenshot 2024-08-22 at 11 28 45 PM

### Search

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[EMI-1961]: https://artsyproduct.atlassian.net/browse/EMI-1961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/Artwork/SaveButton/SaveButton.tsx [hunk 1]
```diff
@@ -43,31 +54,39 @@ export const SaveButtonBase: React.FC<SaveButtonBaseProps> = ({
...
+            height={BTN_HEIGHT}
+          />
+        ) : (
+          <HeartStrokeIcon
+            title={title}
+            fill="black100"
+            width={BTN_WIDTH}
+            height={BTN_HEIGHT}
+          />
+        )}
```
Comment (erikdstock) [inline src/Components/Artwork/SaveButton/SaveButton.tsx:87]: non-blocking: do these need to be repeated or will clickable automatically wrap whichever icon is rendered?

File: src/Components/Artwork/SaveButton/SaveButton.tsx [hunk 2]
```diff
@@ -17,18 +18,28 @@ export interface SaveButtonProps {
...
   const [isHovered, setIsHovered] = useState(false)
   const title = isSaved ? "Unsave" : "Save"
+  const { lotWatcherCount, lotClosesAt, liveBiddingStarted } =
+    artwork.collectorSignals?.auction || {}
+
+  const shouldDisplayLotCount =
+    lotWatcherCount != null &&
+    lotWatcherCount > 0 &&
+    !liveBiddingStarted &&
+    (!lotClosesAt || new Date(lotClosesAt) >= new Date())
```
Comment (erikdstock) [inline src/Components/Artwork/SaveButton/SaveButton.tsx:42]: Again here i think the count presence+non-zero could be simplified to `!!lotWatcherCount`

Bigger question I have here: Does this count update when the users saves/unsaves? As a user I would expect to see that change after i click the heart, but as a developer I would not expect to see that change reflected in the mutation result because it is processed in a job outside the mutation request.

3 ways I could see this working:
1. I could be wrong and it does just work.
2. We update the artwork store surgically when the save/unsave action happens
3. We just maintain some local state in this component including
  - the count on load (in this case i would avoid the null value altogether and set it like `const lotWatcherCount = artwork.collectorSignals.auction.lotWatcherCount || 0`)
  - whether the user had the artwork saved on load (so whether the count includes the user)
  - a state var for when the user saves/unsaves to +/-1

**Review by erikdstock — COMMENTED** [review]
:+1: A few comments, mostly non-blocking. One question about the behavior of the lot watcher count on save which might not be blocking but would irritate me as a user.

File: src/Components/Artwork/SaveButton/SaveButton.tsx [hunk 3]
```diff
@@ -17,18 +18,27 @@ export interface SaveButtonProps {
...
 }) => {
   const [isHovered, setIsHovered] = useState(false)
   const title = isSaved ? "Unsave" : "Save"
+  const { lotWatcherCount, lotClosesAt, liveBiddingStarted } =
+    artwork.collectorSignals?.auction || {}
+
+  const shouldDisplayLotCount =
+    !!lotWatcherCount &&
+    !liveBiddingStarted &&
+    (!lotClosesAt || new Date(lotClosesAt) >= new Date())
```
Comment (MrSltun) [inline src/Components/Artwork/SaveButton/SaveButton.tsx:41]: **Suggestion**: should we hide this behind a feature flag?

File: src/Components/Artwork/Details.tsx [hunk 4]
```diff
@@ -166,7 +185,6 @@ const CollectorSignalLine: React.FC<DetailsProps> = ({
       color="blue100"
       backgroundColor="blue10"
       px={0.5}
-      marginBottom={0.5}
```
Comment (MrSltun) [inline src/Components/Artwork/Details.tsx:169]: **Question**: was this change intentional? 👀

File: src/Components/Artwork/Details.tsx [hunk 5]
```diff
@@ -175,14 +193,66 @@ const CollectorSignalLine: React.FC<DetailsProps> = ({
   )
 }
 
-const EmptyLine: React.FC = () => {
+const BidTimerLine: React.FC<DetailsProps> = ({
```
Comment (MrSltun) [inline src/Components/Artwork/Details.tsx:196]: **Suggestion**: should we extract this one (and the others) into different components so it's easily readable (and have smaller files)?

**Review by MrSltun — COMMENTED** [review]
Looks good! ✨  I just left a few comments  :)
